Mentor fixes and TGUIification

Gilgaxx opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

Fixes this issue #8150 (the second half, it went stale, but never fixed) where a mentoree who requests an mrat and then clicks on the kick mentor button, the "are you sure" alert would shoup up to the mentor instead of the mentoree by changing a "friend" to "friend.owner" in the alert.

TGUIfies 4 prompts related to mentors (Asking for an Mrat, kicking an mrat from the mentoree perspective, leaving as an mrat from the mentor perspective, the Mrat naming window and the actual mhelps (this used to be bugged)).

Also ACTUALLY brings back the hotpink chat for Msay (#12228 did not fix it)

## Why It's Good For The Game

In order:
Fixes a bug, duh. (Mentors shouldn't be like "nuh uh, I'm staying")
TGUI is nicer looking.
I only heard mentors being sad that we lost the hotpink chat.

## Testing Photographs and Procedure
<!-- Include any screenshots/videos/debugging steps of the modified code functioning successfully, ideally including edge cases. -->
<details>
<summary>Screenshots&Videos</summary>
Loaded in with 2 seperate clients to try this out:
Asking for an Mrat:
<img width="640" alt="dreamseeker_O1fcvoojgg" src="https://github.com/user-attachments/assets/d3959e71-c1e2-4401-ae1b-85d7fc03dabe" />

The Mrat naming:
<img width="640" alt="dreamseeker_itL5XPMHIW" src="https://github.com/user-attachments/assets/0109e0b0-1086-4f6d-9ea1-13ef340d44a1" />

Here is the two clients running at the same time (mentor on the right, mentoree on the left)
<img width="1280" alt="dreamseeker_LYvFqhG5Gt" src="https://github.com/user-attachments/assets/bb97628e-c9bd-4507-b91f-6e917fb4829c" />

Leaving as the mentor is now TGUI:
<img width="1280" alt="dreamseeker_VIcvn2p7Ft" src="https://github.com/user-attachments/assets/fa365595-5733-43e2-abf6-253ea252207c" />

Kicking the Mrat as the mentoree is now on TGUI and it's on the right client (tabbing out and back in showed that this prompt was on the right client):
<img width="1280" alt="chrome_Q4jPGSfu2c" src="https://github.com/user-attachments/assets/bc9c6b2f-3a73-4558-a514-3767636dac11" />

The TGUI mhelp (it used to be a tall white box with "Mentorhelp "text"" at the top):
<img width="1280" alt="dreamseeker_MsTvFfAbDV" src="https://github.com/user-attachments/assets/3b356f51-bf48-4536-ad2f-818646a4672f" />
(it still works)
<img width="488" alt="dreamseeker_dCOcN5skHO" src="https://github.com/user-attachments/assets/084dc7c5-6ce8-4a09-b0d2-e81f491ba0bc" />

Disabled local_host_rank in the configs, added myself to the mentor.txt and this is the msay chat now:
<img width="185" alt="dreamseeker_k4TR4l6qX2" src="https://github.com/user-attachments/assets/d787fa1a-c945-4600-9303-17e9f1983e45" />


</details>

## Changelog
:cl:
fix: Mentors can no longer go "nuh uh" when a mentoree tries to kick out their Mrat.
tweak: TGUIfied most mentor related prompts.
tweak: Made Msay for mentors hot pink again!
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Gilgaxx committed:
Update mrat.dm (fixes wrong user alert to remove mrat)

changes a "friend" to "friend.owner" in the alert.

itsmeow commented on code/datums/brain_damage/mrat.dm:
```diff
@@ -186,7 +186,7 @@ CREATION_TEST_IGNORE_SUBTYPES(/mob/camera/imaginary_friend/mrat)
 	if(!istype(friend))
 		qdel(src)
-	if(!istype(friend) || alert(friend, "Are you sure you want to remove your mentor?", "Remove:", "Yes", "No") != "Yes")
+	if(!istype(friend) || alert(friend.owner, "Are you sure you want to remove your mentor?", "Remove:", "Yes", "No") != "Yes")
```
```suggestion
	if(!istype(friend) || tgui_alert(friend.owner, "Are you sure you want to remove your mentor?", "Remove:", list("Yes", "No")) != "Yes")
```

Gilgaxx committed:
TGUIfied some of the mentor verbs

changed some inputs related to mentor actions from the old style to TGUI (the prompt or both mentor and mentoree to leave/kick the mrat, the name requrest for mrat on the mentor side, and the actual mhelp which seemed to be bugged).

Gilgaxx committed:
Makes msay chat hotpink again

Just as the title says

itsmeow commented on code/modules/admin/verbs/mentorhelp.dm:
```diff
@@ -31,12 +31,8 @@ GLOBAL_DATUM_INIT(mhelp_tickets, /datum/help_tickets/mentor, new)
 	var/msg = tgui_input_text(src, "Please describe your problem concisely and a mentor will help as soon as they're able. Remember: Mentors cannot see you or what you're doing. Describe the problem in full detail.", "Mentorhelp contents", multiline = TRUE, encode = FALSE) // we don't encode/sanitize here bc it's done for us later
-	mentorhelp(msg)
-
-/client/verb/mentorhelp(msg as message)
```
Just make sure this is consistent with ahelp. The reason the msg argument exists is so you can type in the chat box instead of clicking the verb

Gilgaxx replied:
You know that's a good point, which made me also aware that the ahelp prompt is ALSO broken and not TGUI 😅
I'll try to come up with a way to fix the TGUI while still maintaining the functionality of beng able to type the message in the verb bar (and probably also apply it to ahelp too).

Gilgaxx replied:
After some more time spent on this, I've noticed this is a much bigger problem, there is not a single verb on the top right that displays the TGUI on the top right. I'm also not great at coding myself and can't think of a solution that fixes this issue while leaving the funcionality of being able to type the message in the verb bar intact. I'll keep trying, but I'd apriciate some help, or I might be forced to drop this part of the PR, which would be a shame.

Gilgaxx committed:
Undoes the TGUI fix for the mentorhelp verb

It's too big in scope for this PR.

Gilgaxx committed:
Merge branch 'master' into mentor-fixes
